homeassistant/components/honeywell/sensor.py

```diff
@@ -0,0 +1,93 @@
+"""Support for Honeywell (US) Total Connect Comfort sensors."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any
+
+from somecomfort import Device
+
+from homeassistant.components.sensor import (
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import (
+    DEVICE_CLASS_HUMIDITY,
+    DEVICE_CLASS_TEMPERATURE,
+    PERCENTAGE,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT,
+)
+from homeassistant.helpers.typing import StateType
+
+from .const import DOMAIN, HUMIDITY_STATUS_KEY, TEMPERATURE_STATUS_KEY
+
+
+@dataclass
+class HoneywellSensorEntityDescriptionMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[[Device], Any]
+    unit_fn: Callable[[Device], Any]
+
+
+@dataclass
+class HoneywellSensorEntityDescription(
+    SensorEntityDescription, HoneywellSensorEntityDescriptionMixin
+):
+    """Describes a Whois sensor entity."""
+
+
+SENSOR_TYPES: tuple[HoneywellSensorEntityDescription, ...] = (
+    HoneywellSensorEntityDescription(
+        key=TEMPERATURE_STATUS_KEY,
+        name="Temperature",
+        device_class=DEVICE_CLASS_TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda device: device.outdoor_temperature,
+        unit_fn=lambda device: TEMP_CELSIUS
+        if device.temperature_unit == "C"
+        else TEMP_FAHRENHEIT,
+    ),
+    HoneywellSensorEntityDescription(
+        key=HUMIDITY_STATUS_KEY,
+        name="Humidity",
+        device_class=DEVICE_CLASS_HUMIDITY,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda device: device.outdoor_humidity,
+        unit_fn=lambda device: PERCENTAGE,
+    ),
+)
+
+
+async def async_setup_entry(hass, config, async_add_entities, discovery_info=None):
+    """Set up the Honeywell thermostat."""
+    data = hass.data[DOMAIN][config.entry_id]
+    sensors = []
+
+    for device in data.devices.values():
```

Comment:
This could be a comprehension but personal preference if it's better or not since it's a double

Reply:
There may be additional sensor types to add in the future. I know that Honeywell also sell indoor sensors, but they aren't compatible with my current system. Perhaps we keep this as is for future proofing?